proof-producer/bin/proof-producer/include/nil/proof-generator/commands/verify_aggregated_command.hpp

```diff
@@ -0,0 +1,215 @@
+                AggregatedFRIVerifier(
+                    PlaceholderConfig config,
+                    std::vector<resources::resource_provider<ConstraintSystem>*>& constraint_system_providers,
+                    std::vector<resources::resource_provider<TableDescription>*>& desc_providers,
+                    std::vector<resources::resource_provider<CommonData>*>& common_data_providers,
+                    resources::resource_provider<AggregatedProof>& aggregated_proof_provider
+                ) : commitment_scheme_fac_(config)
+                  , N(constraint_system_providers.size())
+                {
+                    using resources::subscribe_value;
+                    for (size_t i = 0; i < N; ++i) {
+                        subscribe_value<ConstraintSystem>(*constraint_system_providers[i], constraint_systems_[i]);
+                        subscribe_value<TableDescription>(*desc_providers[i], table_descriptions_[i]);
+                        subscribe_value<CommonData>(*common_data_providers[i], common_datas_[i]);
+                    }
+                    subscribe_value<AggregatedProof>(aggregated_proof_provider, agg_proof_);
+                }
+
+                CommandResult execute() override {
+                    BOOST_ASSERT(agg_proof_);
+                    for (size_t i = 0; i < N; ++i) { 
+                        BOOST_ASSERT(common_datas_[i]);
+                        BOOST_ASSERT(constraint_systems_[i]);
+                        BOOST_ASSERT(table_descriptions_[i]);
+                    }
+
+                    BOOST_LOG_TRIVIAL(info) << "Verifying aggreated dFRI proof...";
+
+                    std::vector<std::shared_ptr<LpcScheme>> lpc_schemes;
+                    for (size_t i = 0; i < N; ++i) {
+                        BOOST_ASSERT(table_descriptions_[i]->rows_amount == table_descriptions_[0]->rows_amount);
+
+                        auto lpc_scheme = commitment_scheme_fac_.make_lpc_scheme(table_descriptions_[i]->rows_amount);
+                        lpc_schemes.push_back(lpc_scheme);
+                    }
+                    // We need to convert circuit<constraint_system> to constraint_system, otherwise it does not convert automatically
+                    // for shared pointers.
+                    std::vector<std::shared_ptr<ConstraintSystemNoCircuit>> constraint_systems;
+                    for (size_t i = 0; i < N; ++i) {
+                        constraint_systems.push_back(std::make_shared<ConstraintSystemNoCircuit>(*constraint_systems_[i]));
+                    }
+
+                    bool verification_result = nil::crypto3::zk::snark::placeholder_DFRI_verifier<BlueprintField, PlaceholderParams>::process(
+                            common_datas_,
+                            *agg_proof_,
+                            table_descriptions_,
+                            constraint_systems,
+                            lpc_schemes,
+                            {} // public inputs
```

Comment:
Here I skip verification of public inputs matching the hashes in common_data. It was done this way in normal verifier, so I also did this way. Not sure if we should read and verify the public inputs for dFRI.